docs/060-breaking-changes.rst

```diff
@@ -9,28 +9,42 @@ For the full list check
 `the release changelog <https://github.com/ethereum/solidity/releases/tag/v0.6.0>`_.
 
 
-Syntactic Only Changes
-======================
+Changes the Compiler Might not Warn About
+=========================================
 
-This section lists purely syntactic changes that do not affect the behavior of existing code.
+This section lists changes where the behaviour of your code might
+change without the compiler telling you about it.
 
-* Conversions from external function types to ``address`` are now disallowed. Instead external
-  function types have a member called ``address``, similar to the existing ``selector`` member.
-* Conversions from ``address`` to ``address payable`` are now possible via ``payable(x)``, where
-  ``x`` must be of type ``address``.
+* The resulting type of an exponentiation is the type of the base. It used to be the smallest type
+  that can hold both the type of the base and the type of the exponent, as with symmentric
+  operations. Additionally, signed types are allowed for the base of the exponetation.
 
-* Function ``push(value)`` for dynamic storage arrays do not return the new length anymore.
 
-* The new keyword ``abstract`` can be used to mark contracts as abstract. It has to be used
-  if a contract does not implement all its functions.
+Explicitness Requirements
+=========================
 
-* Libraries have to implement all their functions, not only the internal ones.
+This section lists changes where the code now needs to be more explicit,
+but the semantics to do not change.
+For most of the topics the compiler will provide suggestions.
+
+* Functions can now only be overridden when they are either marked with the
+  ``virtual`` keyword or defined in an interface. Functions without
+  implementation outside an interface have to be marked ``virtual``.
+  When overriding a function or modifier, the new keyword ``override``
+  must be used. When overriding a function or modifier defined in multiple
+  parallel bases, all bases must be listed in parentheses after the keyword
+  like so: ``override(Base1, Base2)``.
 
-* Member-access to ``length`` of arrays is now always read-only, even for storage arrays. It's no
+* Member-access to ``length`` of arrays is now always read-only, even for storage arrays. It is no
   longer possible to resize storage arrays assigning a new value to their length. Use ``push()``,
   ``push(value)`` or ``pop()`` instead, or assign a full array, which will of course overwrite existing content.
```

Comment:
Unrelated to this PR: Are the following examples equivalent?

Push:
```
contract C { bool[] array; function foo() public { array.push(true); }
```

Indexing:
```
contract C { bool[] array; function foo() public { array[0] = true; }
```

Reply:
The second will revert because `array.length = 0`.